### PR TITLE
Reference texture support

### PR DESCRIPTION
**Pull request by Qainguin**
Added Reference Image Button
Screenshot 2024-05-10 at 7 51 35 PM

Reference Popup
Screenshot 2024-05-10 at 7 53 31 PM

No Overlay
Screenshot 2024-05-10 at 7 53 15 PM

Overlay
Screenshot 2024-05-10 at 7 53 50 PM

### CODE REVIEW

File: src/GlobalSettings.gd
Comment (Qainguin) [inline src/GlobalSettings.gd:1]: I don't like that I have to use a signal in GlobalSettings, it feels like I'm doing something wrong.

**Comment by MewPurPur** [comment]
This is an excellent initial implementation.

**Comment by Qainguin** [comment]
Thanks! Should I merge or wait?

File: src/GlobalSettings.gd
Comment (Qainguin) [reply src/GlobalSettings.gd:1]: Ok. I think I'll go figure that out.

**Comment by MewPurPur** [comment]
Mark as ready if you want me to review it.

**Comment by MewPurPur** [comment]
Thanks!

**Pull request merged**